WireDomain/Sources/WireDomain/Event Processing/FederationEventProcessor/FederationConnectionRemovedEventProcessor.swift

```diff
@@ -32,9 +32,20 @@ protocol FederationConnectionRemovedEventProcessorProtocol {
 
 struct FederationConnectionRemovedEventProcessor: FederationConnectionRemovedEventProcessorProtocol {
 
-    func processEvent(_: FederationConnectionRemovedEvent) async throws {
-        // TODO: [WPB-10187]
-        assertionFailure("not implemented yet")
+    enum Error: Swift.Error {
+        case missingDomains(Set<String>)
+    }
+
+    let repository: any ConnectionsRepositoryProtocol
+
+    func processEvent(_ event: FederationConnectionRemovedEvent) async throws {
+        guard let domain = Array(event.domains).first,
+              let otherDomain = Array(event.domains).last
+        else {
+            throw Error.missingDomains(event.domains)
+        }
```

Comment:
Since `event.domains` is a `Set`, the order is not defined when it is converted to an `Array`, which means it's possible that `domain` and `otherDomain` are the same. I think we should also validate that there are 2 domains (if that is indeed the case).

```suggestion
        let domains = Array(event.domains)
        guard 
            domains.count == 2,
            let domain = domains.first,
            let otherDomain = domains.last
        else {
            throw Error.missingDomains(event.domains)
        }
```

Reply:
fixed thx